core/test/qa_buffer.cpp

```diff
@@ -22,13 +22,13 @@ struct TestStruct {
     [[nodiscard]] constexpr bool test() const noexcept { return true; }
 };
 
-void consumableInputRangeTest1(auto in) { [[maybe_unused]] const auto inFirst = in[0]; }
+void consumableInputRangeTest1(auto in) { [[maybe_unused]] volatile auto isEmpty = in.empty(); }
 
-void consumableInputRangeTest2(gr::ConsumableSpan auto in) { [[maybe_unused]] const auto inFirst = in[0]; }
+void consumableInputRangeTest2(gr::ConsumableSpan auto in) { [[maybe_unused]] volatile auto isEmpty = in.empty(); }
 
 template<typename T>
 void consumableInputRangeTest3(std::span<const T> in) {
-    [[maybe_unused]] const auto inFirst = in[0];
```

Comment:
What is this testing anyway? The previous code seems to test that `operator[]` exists. The new test is whether `empty()` exists? How useful is this? To me that looks like something you'd rather `static_assert` and then these tests can be removed altogether?

Reply:
I think the main idea of these test is to test the acceptable signatures of the function.

Reply:
I think these tests can be removed, there are similar test in `qa_Block` which test different `processBulk` signatures.

Reply:
LGTM :+1:

Reply:
Could you please approve the PR? Thank you!